dist/bin/scalac

```diff
@@ -30,43 +30,42 @@ source "$PROG_HOME/bin/common"
 
 [ -z "$PROG_NAME" ] && PROG_NAME=$CompilerMain
 
-withCompiler=true
-
 while [[ $# -gt 0 ]]; do
-case "$1" in
-           --) shift; for arg; do addResidual "$arg"; done; set -- ;;
-  -v|-verbose) verbose=true && addScala "-verbose" && shift ;;
-    -q|-quiet) quiet=true && shift ;;
-
-    # Optimize for short-running applications, see https://github.com/lampepfl/dotty/issues/222
-    -Oshort) addJava "-XX:+TieredCompilation -XX:TieredStopAtLevel=1" && shift ;;
-        -repl) PROG_NAME="$ReplMain" && shift ;;
-      -script) PROG_NAME="$ScriptingMain" && target_script="$2" && shift && shift
-               while [[ $# -gt 0 ]]; do addScript "$1" && shift ; done ;;
-     -compile) PROG_NAME="$CompilerMain" && shift ;;
-   -decompile) PROG_NAME="$DecompilerMain" && shift ;;
- -print-tasty) PROG_NAME="$TastyPrinterMain" && shift ;;
-         -run) PROG_NAME="$ReplMain" && shift ;;
-      -colors) colors=true && shift ;;
-   -no-colors) unset colors && shift ;;
-  -with-compiler) jvm_cp_args="$PSEP$DOTTY_COMP$PSEP$TASTY_CORE" && shift ;;
-
-          # break out -D and -J options and add them to java_args so
-          # they reach the JVM in time to do some good. The -D options
-          # will be available as system properties.
-          -D*) addJava "$1" && shift ;;
-          -J*) addJava "${1:2}" && shift ;;
-            *) addResidual "$1" && shift ;;
+  case "$1" in
+    -D*)
+      # pass to scala as well: otherwise we lose it sometimes when we
+      # need it, e.g. communicating with a server compiler.
+      addJava "$1"
+      addScala "$1"
+      # respect user-supplied -Dscala.usejavacp
+      shift
+      ;;
+    -J*)
+      # as with -D, pass to scala even though it will almost
+      # never be used.
+      addJava "${1:2}"
+      addScala "$1"
+      shift
+      ;;
+    -classpath*)
```

Comment:
Same confusion here. This did not use to be special in the old script. Why is it special now?

Reply:
This is copied from the Scala script, I think there was the same Cygwin bug there